pythnet/remote-executor/programs/remote-executor/src/state/governance_payload.rs

```diff
@@ -127,10 +146,130 @@ impl ExecutorPayload {
         (self.header.magic_number == MAGIC_NUMBER)
             .ok_or(error!(ExecutorError::GovernanceHeaderInvalidMagicNumber))?;
         (self.header.module == ExecutorPayload::MODULE)
-            .ok_or(error!(ExecutorError::GovernanceHeaderInvalidMagicNumber))?;
+            .ok_or(error!(ExecutorError::GovernanceHeaderInvalidModule))?;
         (self.header.action == ExecutorPayload::ACTION)
-            .ok_or(error!(ExecutorError::GovernanceHeaderInvalidMagicNumber))?;
+            .ok_or(error!(ExecutorError::GovernanceHeaderInvalidAction))?;
         (Chain::from(self.header.chain.value) == Chain::Pythnet)
-            .ok_or(error!(ExecutorError::GovernanceHeaderInvalidMagicNumber))
+            .ok_or(error!(ExecutorError::GovernanceHeaderInvalidReceiverChain))
+    }
+}
+
+#[cfg(test)]
+pub mod tests {
+    use crate::{
+        error,
+        error::ExecutorError,
+        state::governance_payload::InstructionData,
+    };
+
+    use super::{
+        Action,
+        BigEndianU16,
+        ExecutorPayload,
+        Module,
+        MAGIC_NUMBER,
+    };
+    use anchor_lang::{
+        prelude::Pubkey,
+        AnchorDeserialize,
+        AnchorSerialize,
+    };
+    use wormhole::Chain;
+
+    #[test]
+    fn test_check_deserialization_serialization() {
+        // No instructions
+        let payload = ExecutorPayload {
+            header: super::GovernanceHeader {
+                magic_number: MAGIC_NUMBER,
+                module: Module::Executor,
+                action: Action::ExecutePostedVaa,
+                chain: BigEndianU16 {
+                    value: Chain::Pythnet.try_into().unwrap(),
+                },
+            },
+            instructions: vec![],
+        };
+
+        assert!(payload.check_header().is_ok());
+
+        let payload_bytes = payload.try_to_vec().unwrap();
+        assert_eq!(payload_bytes, vec![80, 84, 71, 77, 0, 0, 0, 26, 0, 0, 0, 0]);
```

Comment:
The first part of this seems fine
[80, 84, 71, 77] -> MAGIC ✅ 
[0] -> Module Executor ✅ 
[0] -> Action ExecutePostedVaa ✅  
[0, 26] -> Big endian chain id of Pythnet ✅ 

I guess the 4 other ones are related to borsh deserialization of empty vector (size, maybe?)

Reply:
Yes ser that's it